### PR TITLE
Refactor AddressBook to GuestBook

### PR DESCRIPTION
**Pull request by bryanhce**
Also refactored Person to Guest.

Did not refactor test cases.

### CODE REVIEW

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -195,19 +195,19 @@ Step 1. The user launches the application for the first time. The `VersionedAddr
 
 ![UndoRedoState0](images/UndoRedoState0.png)
 
-Step 2. The user executes `delete 5` command to delete the 5th person in the address book. The `delete` command calls `Model#commitAddressBook()`, causing the modified state of the address book after the `delete 5` command executes to be saved in the `addressBookStateList`, and the `currentStatePointer` is shifted to the newly inserted address book state.
+Step 2. The user executes `delete 5` command to delete the 5th guest in the address book. The `delete` command calls `Model#commitAddressBook()`, causing the modified state of the address book after the `delete 5` command executes to be saved in the `addressBookStateList`, and the `currentStatePointer` is shifted to the newly inserted address book state.
```
Comment (jehousoh) [inline docs/DeveloperGuide.md]: Should the address book be changed to guest book?

File: docs/DeveloperGuide.md [hunk 2]
```diff
@@ -195,19 +195,19 @@ Step 1. The user launches the application for the first time. The `VersionedAddr
...
 
 ![UndoRedoState1](images/UndoRedoState1.png)
 
-Step 3. The user executes `add n/David …​` to add a new person. The `add` command also calls `Model#commitAddressBook()`, causing another modified address book state to be saved into the `addressBookStateList`.
+Step 3. The user executes `add n/David …​` to add a new guest. The `add` command also calls `Model#commitAddressBook()`, causing another modified address book state to be saved into the `addressBookStateList`.
```
Comment (jehousoh) [inline docs/DeveloperGuide.md]: Should this address book be changed to guest book?

File: docs/DeveloperGuide.md [hunk 3]
```diff
@@ -195,19 +195,19 @@ Step 1. The user launches the application for the first time. The `VersionedAddr
...
 
 ![UndoRedoState2](images/UndoRedoState2.png)
 
 <div markdown="span" class="alert alert-info">:information_source: **Note:** If a command fails its execution, it will not call `Model#commitAddressBook()`, so the address book state will not be saved into the `addressBookStateList`.
```
Comment (jehousoh) [inline docs/DeveloperGuide.md]: Should this address book be changed to guest book?

File: docs/DeveloperGuide.md [hunk 4]
```diff
@@ -195,19 +195,19 @@ Step 1. The user launches the application for the first time. The `VersionedAddr
...
 
 </div>
 
-Step 4. The user now decides that adding the person was a mistake, and decides to undo that action by executing the `undo` command. The `undo` command will call `Model#undoAddressBook()`, which will shift the `currentStatePointer` once to the left, pointing it to the previous address book state, and restores the address book to that state.
+Step 4. The user now decides that adding the guest was a mistake, and decides to undo that action by executing the `undo` command. The `undo` command will call `Model#undoAddressBook()`, which will shift the `currentStatePointer` once to the left, pointing it to the previous address book state, and restores the address book to that state.
```
Comment (jehousoh) [inline docs/DeveloperGuide.md]: Should this address book be changed to guest book?

File: docs/DeveloperGuide.md [hunk 5]
```diff
@@ -302,11 +302,11 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
...
-| `* * *`  | user                                       | find a person by name          | locate details of persons without having to go through the entire list |
+| `* * *`  | user                                       | add a new guest               |                                                                        |
+| `* * *`  | user                                       | delete a guest                | remove entries that I no longer need                                   |
+| `* * *`  | user                                       | find a guest by name          | locate details of guests without having to go through the entire list |
 | `* *`    | user                                       | hide private contact details   | minimize chance of someone else seeing them by accident                |
-| `*`      | user with many persons in the address book | sort persons by name           | locate a person easily                                                 |
+| `*`      | user with many guests in the address book | sort guests by name           | locate a guest easily                                                 |
```
Comment (jehousoh) [inline docs/DeveloperGuide.md]: Should this address book be changed to guest book?

File: docs/diagrams/ArchitectureSequenceDiagram.puml [hunk 6]
```diff
@@ -19,7 +19,7 @@ activate model MODEL_COLOR
 model -[MODEL_COLOR]-> logic
 deactivate model
 
-logic -[LOGIC_COLOR]> storage : saveAddressBook(addressBook)
+logic -[LOGIC_COLOR]> storage : saveAddressBook(guestBook)
```
Comment (jehousoh) [inline docs/diagrams/ArchitectureSequenceDiagram.puml]: Should saveAddressBook be changed to saveGuestBook?

File: docs/tutorials/AddRemark.md [hunk 7]
```diff
@@ -23,12 +23,12 @@ For now, let’s keep `RemarkCommand` as simple as possible and print some outpu
...
 ``` java
-package seedu.address.logic.commands;
+package seedu.guest.logic.commands;
 
-import seedu.address.model.Model;
+import seedu.guest.model.Model;
 
 /**
- * Changes the remark of an existing person in the address book.
+ * Changes the remark of an existing guest in the address book.
```
Comment (jehousoh) [inline docs/tutorials/AddRemark.md:31]: Should this address book be changed to guest book?

File: docs/tutorials/TracingCode.md [hunk 8]
```diff
@@ -231,7 +231,7 @@ Recall from the User Guide that the `edit` command has the format: `edit INDEX [
      * {@code JsonSerializableAddressBook}.
      */
     public JsonSerializableAddressBook(ReadOnlyAddressBook source) {
```
Comment (jehousoh) [inline docs/tutorials/TracingCode.md]: Should the JsonSerializableAddressBook(ReadOnlyAddressBook source) be changed to JsonSerializableGuestBook(ReadOnlyGuestBook source)?

File: docs/tutorials/RemovingFields.md [hunk 9]
```diff
@@ -28,7 +28,7 @@ IntelliJ IDEA provides a refactoring tool that can identify *most* parts of a re
 
 ### Assisted refactoring
 
-The `address` field in `Person` is actually an instance of the `seedu.address.model.person.Address` class. Since removing the `Address` class will break the application, we start by identifying `Address`'s usages. This allows us to see code that depends on `Address` to function properly and edit them on a case-by-case basis. Right-click the `Address` class and select `Refactor` \> `Safe Delete` through the menu.
+The `address` field in `Person` is actually an instance of the `seedu.guest.model.guest.Address` class. Since removing the `Address` class will break the application, we start by identifying `Address`'s usages. This allows us to see code that depends on `Address` to function properly and edit them on a case-by-case basis. Right-click the `Address` class and select `Refactor` \> `Safe Delete` through the menu.
```
Comment (jehousoh) [inline docs/tutorials/RemovingFields.md]: Should Person be changed to Guest?

File: docs/tutorials/RemovingFields.md [hunk 10]
```diff
@@ -100,7 +100,7 @@ In `src/test/data/`, data meant for testing purposes are stored. While keeping t
 
 ```json
 {
-  "persons": [ {
+  "guests": [ {
     "name": "Person with invalid name field: Ha!ns Mu@ster",
```
Comment (jehousoh) [inline docs/tutorials/RemovingFields.md]: Should Person be changed to Guest?

File: docs/tutorials/TracingCode.md [hunk 11]
```diff
@@ -189,22 +189,22 @@ Recall from the User Guide that the `edit` command has the format: `edit INDEX [
    @Override
    public CommandResult execute(Model model) throws CommandException {
        ...
-       Person personToEdit = lastShownList.get(index.getZeroBased());
-       Person editedPerson = createEditedPerson(personToEdit, editPersonDescriptor);
-       if (!personToEdit.isSamePerson(editedPerson) && model.hasPerson(editedPerson)) {
+       Person guestToEdit = lastShownList.get(index.getZeroBased());
+       Person editedGuest = createEditedPerson(guestToEdit, editGuestDescriptor);
+       if (!guestToEdit.isSamePerson(editedGuest) && model.hasPerson(editedGuest)) {
```
Comment (jehousoh) [inline docs/tutorials/TracingCode.md]: Should Person be changed to Guest?

File: docs/tutorials/TracingCode.md [hunk 12]
```diff
@@ -189,22 +189,22 @@ Recall from the User Guide that the `edit` command has the format: `edit INDEX [
...
            throw new CommandException(MESSAGE_DUPLICATE_PERSON);
        }
-       model.setPerson(personToEdit, editedPerson);
+       model.setPerson(guestToEdit, editedGuest);
```
Comment (jehousoh) [inline docs/tutorials/TracingCode.md]: Should Person be changed to Guest?

File: docs/tutorials/TracingCode.md [hunk 13]
```diff
@@ -189,22 +189,22 @@ Recall from the User Guide that the `edit` command has the format: `edit INDEX [
...
        model.updateFilteredPersonList(PREDICATE_SHOW_ALL_PERSONS);
-       return new CommandResult(String.format(MESSAGE_EDIT_PERSON_SUCCESS, editedPerson));
+       return new CommandResult(String.format(MESSAGE_EDIT_PERSON_SUCCESS, editedGuest));
```
Comment (jehousoh) [inline docs/tutorials/TracingCode.md]: Should Person be changed to Guest?

File: docs/tutorials/TracingCode.md [hunk 14]
```diff
@@ -189,22 +189,22 @@ Recall from the User Guide that the `edit` command has the format: `edit INDEX [
...
-   * it uses the `setPerson()` method (defined in the interface `Model` and implemented in `ModelManager` as per the usual pattern) to update the person data.
-   * it uses the `updateFilteredPersonList` method to ask the `Model` to populate the 'filtered list' with _all_ persons.<br>
-     FYI, The 'filtered list' is the list of persons resulting from the most recent operation that will be shown to the user immediately after. For the `edit` command, we populate it with all the persons so that the user can see the edited person along with all other persons. If this was a `find` command, we would be setting that list to contain the search results instead.<br>
-     To provide some context, given below is the class diagram of the `Model` component. See if you can figure out where the 'filtered list' of persons is being tracked.
+   * it uses the `setPerson()` method (defined in the interface `Model` and implemented in `ModelManager` as per the usual pattern) to update the guest data.
+   * it uses the `updateFilteredPersonList` method to ask the `Model` to populate the 'filtered list' with _all_ guests.<br>
```
Comment (jehousoh) [inline docs/tutorials/TracingCode.md]: Should setPerson and updateFilteredPersonList be changed to setGuest and updateFilteredGuestList?

File: docs/tutorials/TracingCode.md [hunk 15]
```diff
@@ -231,7 +231,7 @@ Recall from the User Guide that the `edit` command has the format: `edit INDEX [
      * {@code JsonSerializableAddressBook}.
      */
     public JsonSerializableAddressBook(ReadOnlyAddressBook source) {
-        persons.addAll(
+        guests.addAll(
             source.getPersonList()
                   .stream()
                   .map(JsonAdaptedPerson::new)
```
Comment (jehousoh) [inline docs/tutorials/TracingCode.md]: should getPersonList and JsonAdaptedPerson be changed to getGuestList and JsonAdaptedGuest?

File: docs/tutorials/AddRemark.md [hunk 16]
```diff
@@ -343,25 +343,25 @@ save it with `Model#setPerson()`.
...
-                personToEdit.getName(), personToEdit.getPhone(), personToEdit.getEmail(),
-                personToEdit.getAddress(), remark, personToEdit.getTags());
+        Person guestToEdit = lastShownList.get(index.getZeroBased());
+        Person editedGuest = new Person(
+                guestToEdit.getName(), guestToEdit.getPhone(), guestToEdit.getEmail(),
+                guestToEdit.getAddress(), remark, guestToEdit.getTags());
 
-        model.setPerson(personToEdit, editedPerson);
+        model.setPerson(guestToEdit, editedGuest);
         model.updateFilteredPersonList(PREDICATE_SHOW_ALL_PERSONS);
```
Comment (jehousoh) [inline docs/tutorials/AddRemark.md]: Should Person be changed to Guest?

File: docs/tutorials/AddRemark.md [hunk 17]
```diff
@@ -343,25 +343,25 @@ save it with `Model#setPerson()`.
             throw new CommandException(Messages.MESSAGE_INVALID_PERSON_DISPLAYED_INDEX);
         }
 
-        Person personToEdit = lastShownList.get(index.getZeroBased());
-        Person editedPerson = new Person(
-                personToEdit.getName(), personToEdit.getPhone(), personToEdit.getEmail(),
-                personToEdit.getAddress(), remark, personToEdit.getTags());
+        Person guestToEdit = lastShownList.get(index.getZeroBased());
+        Person editedGuest = new Person(
```
Comment (jehousoh) [inline docs/tutorials/AddRemark.md]: Should Person be changed to Guest?

File: docs/tutorials/AddRemark.md [hunk 18]
```diff
@@ -343,25 +343,25 @@ save it with `Model#setPerson()`.
...
     }
 
     /**
      * Generates a command execution success message based on whether
      * the remark is added to or removed from
-     * {@code personToEdit}.
+     * {@code guestToEdit}.
      */
-    private String generateSuccessMessage(Person personToEdit) {
+    private String generateSuccessMessage(Person guestToEdit) {
```
Comment (jehousoh) [inline docs/tutorials/AddRemark.md]: Should Person be changed to Guest?

File: docs/tutorials/AddRemark.md [hunk 19]
```diff
@@ -311,9 +311,9 @@ Just add [this one line of code!](https://github.com/se-edu/addressbook-level3/c
 **`PersonCard.java`:**
 
 ``` java
-public PersonCard(Person person, int displayedIndex) {
+public PersonCard(Person guest, int displayedIndex) {
```
Comment (jehousoh) [inline docs/tutorials/AddRemark.md]: Should Person be changed to Guest?

File: docs/tutorials/AddRemark.md [hunk 20]
```diff
@@ -225,11 +225,11 @@ If you are stuck, check out the sample
 
 ## Add `Remark` to the model
 
-Now that we have all the information that we need, let’s lay the groundwork for propagating the remarks added into the in-memory storage of person data. We achieve that by working with the `Person` model. Each field in a Person is implemented as a separate class (e.g. a `Name` object represents the person’s name). That means we should add a `Remark` class so that we can use a `Remark` object to represent a remark given to a person.
+Now that we have all the information that we need, let’s lay the groundwork for propagating the remarks added into the in-memory storage of guest data. We achieve that by working with the `Person` model. Each field in a Person is implemented as a separate class (e.g. a `Name` object represents the guest’s name). That means we should add a `Remark` class so that we can use a `Remark` object to represent a remark given to a guest.
```
Comment (jehousoh) [inline docs/tutorials/AddRemark.md]: Should Person be changed to Guest?

**Review by jehousoh — CHANGES_REQUESTED** [review]

File: docs/DeveloperGuide.md [hunk 21]
```diff
@@ -118,7 +118,7 @@ Here's a (partial) class diagram of the `Logic` component:
 How the `Logic` component works:
 1. When `Logic` is called upon to execute a command, it uses the `AddressBookParser` class to parse the user command.
```
Comment (kangzongxian) [inline docs/DeveloperGuide.md]: Should this be GuestBookParser, or do we not consider refactoring the classes yet? Similar cases at e.g Line 198, etc... about changing AddressBook -> GuestBook?

File: docs/DeveloperGuide.md [hunk 22]
```diff
@@ -436,17 +436,17 @@ testers are expected to do more *exploratory* testing.
...
+### Deleting a guest
 
-1. Deleting a person while all persons are being shown
+1. Deleting a guest while all guests are being shown
 
-   1. Prerequisites: List all persons using the `list` command. Multiple persons in the list.
+   1. Prerequisites: List all guests using the `list` command. Multiple guests in the list.
 
    1. Test case: `delete 1`<br>
       Expected: First contact is deleted from the list. Details of the deleted contact shown in the status message. Timestamp in the status bar is updated.
```
Comment (kangzongxian) [inline docs/DeveloperGuide.md]: Should this be guest as well? Changing contact -> guest

File: docs/DeveloperGuide.md [hunk 23]
```diff
@@ -302,11 +302,11 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
...
-| `* * *`  | user                                       | add a new person               |                                                                        |
-| `* * *`  | user                                       | delete a person                | remove entries that I no longer need                                   |
-| `* * *`  | user                                       | find a person by name          | locate details of persons without having to go through the entire list |
+| `* * *`  | user                                       | add a new guest               |                                                                        |
+| `* * *`  | user                                       | delete a guest                | remove entries that I no longer need                                   |
+| `* * *`  | user                                       | find a guest by name          | locate details of guests without having to go through the entire list |
 | `* *`    | user                                       | hide private contact details   | minimize chance of someone else seeing them by accident                |
```
Comment (kangzongxian) [inline docs/DeveloperGuide.md]: Should we change contact to guest as well?

File: docs/tutorials/AddRemark.md [hunk 24]
```diff
@@ -240,9 +240,9 @@ Let’s change `RemarkCommand` and `RemarkCommandParser` to use the new `Remark`
 
 ## Add a placeholder element for remark to the UI
 
-Without getting too deep into `fxml`, let’s go on a 5 minute adventure to get some placeholder text to show up for each person.
+Without getting too deep into `fxml`, let’s go on a 5 minute adventure to get some placeholder text to show up for each guest.
 
-Simply add the following to [`seedu.address.ui.PersonCard`](https://github.com/se-edu/addressbook-level3/commit/850b78879582f38accb05dd20c245963c65ea599#diff-639834f1e05afe2276a86372adf0fe5f69314642c2d93cfa543d614ce5a76688).
+Simply add the following to [`seedu.guest.ui.GuestCard`](https://github.com/se-edu/addressbook-level3/commit/850b78879582f38accb05dd20c245963c65ea599#diff-639834f1e05afe2276a86372adf0fe5f69314642c2d93cfa543d614ce5a76688).
 
 **`PersonCard.java`:**
```
Comment (JustinPeng13) [inline docs/tutorials/AddRemark.md]: Should these 'Person's be 'Guest's?

**Review by JustinPeng13 — CHANGES_REQUESTED** [review]

File: docs/DeveloperGuide.md (hunk 23 above)
Comment (bryanhce) [reply docs/DeveloperGuide.md]: Good catch! I will change that

**Review by jehousoh — APPROVED** [review]
lgtm

**Review by kangzongxian — APPROVED** [review]
Looks good to me now, thank you!

**Review by JustinPeng13 — APPROVED** [review]
LGTM

**Pull request merged**